### ISSUE TITLE
Crash on :undelete if file contains a %-sign

### ISSUE THREAD

**Issue by woefe** [karlch/vimiv-qt#228]
Steps to reproduce:
1. Create an image with filename `%.jpg`
2. `vimiv '%.jpg'`
3. `:delete %`
4. `:undelete`
5. Crash:
```
[18:43:26] ERROR    <vimiv>              Uncaught exception! Exiting gracefully and printing stack...
Traceback (most recent call last):
  File "/home/woefe/workspace/vimiv-qt/vimiv/gui/commandline.py", line 91, in <lambda>
    QTimer.singleShot(0, lambda: runners.run(command, mode=self.mode))
  File "/home/woefe/workspace/vimiv-qt/vimiv/commands/runners.py", line 49, in inner
    return func(text, *args, **kwargs)
  File "/home/woefe/workspace/vimiv-qt/vimiv/commands/runners.py", line 80, in run
    _run_single(cmdpart, mode, count)
  File "/home/woefe/workspace/vimiv-qt/vimiv/commands/runners.py", line 49, in inner
    return func(text, *args, **kwargs)
  File "/home/woefe/workspace/vimiv-qt/vimiv/commands/runners.py", line 99, in _run_single
    command(count + text, mode)
  File "/home/woefe/workspace/vimiv-qt/vimiv/commands/runners.py", line 119, in command
    _run_command(count, cmdname, args, mode)
  File "/home/woefe/workspace/vimiv-qt/vimiv/commands/runners.py", line 154, in _run_command
    cmd(args, count=count)
  File "/home/woefe/workspace/vimiv-qt/vimiv/api/commands.py", line 318, in __call__
    objreg._call_with_instance(self.func, **kwargs)
  File "/home/woefe/workspace/vimiv-qt/vimiv/api/objreg.py", line 73, in _call_with_instance
    return func(*args, **kwargs)
  File "/home/woefe/workspace/vimiv-qt/vimiv/utils/trash_manager.py", line 89, in undelete
    original_filename, _ = trash_info(basename)
  File "/home/woefe/workspace/vimiv-qt/vimiv/utils/trash_manager.py", line 160, in trash_info
    original_filename = content["Path"]
  File "/usr/lib/python3.8/configparser.py", line 1255, in __getitem__
    return self._parser.get(self._name, key)
  File "/usr/lib/python3.8/configparser.py", line 799, in get
    return self._interpolation.before_get(self, section, option, value,
  File "/usr/lib/python3.8/configparser.py", line 395, in before_get
    self._interpolate_some(parser, option, L, value, section, defaults, 1)
  File "/usr/lib/python3.8/configparser.py", line 442, in _interpolate_some
    raise InterpolationSyntaxError(
configparser.InterpolationSyntaxError: '%' must be followed by '%' or '(', found: '%.jpg'
```
---
`vimiv --version`
```
vimiv v0.8.0
Git: 3ccfcc87b00adbb477a6c2b727bc7d5eddfde1be (2020-06-01)

Python: 3.8.3
Qt: 5.15.0
PyQt: 5.15.0

Svg Support: True
Piexif: None

Paths:
cache: /home/woefe/.cache/vimiv
config: /home/woefe/.config/vimiv
data: /home/woefe/.local/share/vimiv
```

**Comment by woefe** [comment]
It is probably enough to disable interpolation in the ConfigParser. It fixes the crash, but I did not check if it introduces any regressions.
```patch
diff --git a/vimiv/utils/trash_manager.py b/vimiv/utils/trash_manager.py
index 2f55c29..d17ad45 100644
--- a/vimiv/utils/trash_manager.py
+++ b/vimiv/utils/trash_manager.py
@@ -154,7 +154,7 @@ def trash_info(filename: str) -> Tuple[str, str]:
         deletion_date: The deletion date.
     """
     info_filename = _get_info_filename(filename)
-    info = configparser.ConfigParser()
+    info = configparser.ConfigParser(interpolation=None)
     info.read(info_filename)
     content = info["Trash Info"]
     original_filename = content["Path"]
```

**Comment by woefe** [comment]
I have been playing around with this some more. A %-sign in the filenames of completions for the `:undelete` command should also be escaped (backslash or double % worked for me).

**Comment by karlch** [comment]
Thanks for another detailed report! Your proposed fix works for me, but I still want to do more sanity checks and possible clean up the code a bit. The completion of course also makes sense, probably in general for filenames concerning the %-sign, not only for `:undelete`.

**Comment by karlch** [comment]
Let me know in case there are remaining issues after #229 :smile:

**Comment by woefe** [comment]
There is still a problem, if the filename is `\%.jpg`. In that case vimiv will print an error on command `:delete %`:
```
delete: No paths matching '/tmp/tmp.SdUnkrUblm/%.jpg'
```
And the %-sign is not properly escaped in completions.

**Comment by karlch** [comment]
The not properly escaped also refers to the filename with a backslash, i.e. `\%.jpg`?

**Comment by woefe** [comment]
Yes, it refers to the filename with both backslash and percent sign.

**Comment by karlch** [comment]
Alright, this makes sense, backslash is already treated as escape char, I will look into this. These filenames are getting out of hand :laughing:

**Comment by karlch** [comment]
I hope this is now fixed once and for all with 63b63f2c35af176ff9b6c2b35033dd67e5cfcf73 and 207ff8f5ee57cf123025c2ab2f009c16fdf692bc. Just let me know in case there are further issues.
